### DISCUSSION TITLE
Channel Certification & SDK Conversion?

### DISCUSSION THREAD

**Discussion by Play-AV** [rrirower/DP2SceneGraphWizard#58] [Q&A]
As DP is being sunset, I've been working on transitioning a channel over to the SDK. I pushed a demo channel with a smaller selection of content about a month ago but it still hasn't been approved. Troublesome but I know Roku is slow with these things occasionally.

I have an existing, obviously approved channel. Does anyone know if I hit convert to SDK and upload my packages whether the Direct Publisher version of the channel will continue to be available until the SDK version is approved? The worry is that I convert, the SDK channel I submit fails certification and then there's no channel at all. 

Cheers

**Comment by rrirower** [comment]
Is your question related to using the wizard software?

**Comment by ferdiworks** [comment]
@Play-AV Once you upload the encrypted package to Roku and Run and pass static  Analysis and deep link you submit for publishing! Your DP Channel will stay active until the SDK channel is published same holds if you upload a new SDK the old channel stays active till the new one is published!

**Reply by Play-AV** [reply]
> Is your question related to using the wizard software?


Yes, somewhat at least, since I'm using the Wizard. 
Turns out my channel had gotten 'stuck', some days of emailing back and forth with Roku got it fixed. Prior to this, my fear was that the code used here was good enough to run in development mode but may fail testing, hence why I posted here. 

Reporting in to say the demo channel got approved using DP2SceneGraphWizard. There is a bug related to series with seasons however, if you have seasons with episodes, playback is problematic (often times on seasons, playback for an episode would just play the first episode in the JSON) 
> @Play-AV Once you upload the encrypted package to Roku and Run and pass static Analysis and deep link you submit for publishing! Your DP Channel will stay active until the SDK channel is published same holds if you upload a new SDK the old channel stays active till the new one is published!

Out of curiosity, do you manually add in your deep linking for every item?  I was trying to go back and see how I had set it up on the channel that was just approved but apparently, you have to upload a new package to see it.

**Reply by rrirower** [reply]
@Play-AV 

`if you have seasons nested within episodes, playback is problematic`

Seasons nested within Episodes are not standard practice.  A Series can own one or more Seasons with nested Episodes, or, a Series can own nested Episodes.  The structure you are trying to use does not conform to the Roku Json Spec.

**Comment by Play-AV** [comment]
> @Play-AV
> 
> `if you have seasons nested within episodes, playback is problematic`
> 
>Seasons nested within Episodes are not standard practice. A Series can own one or more Seasons with nested Episodes, or, a Series can own nested Episodes. The structure you are trying to use does not conform to the Roku Json Spec.


Sorry, that was a typo. I meant to say, series that have seasons with episodes nested.

**Reply by rrirower** [reply]
@Play-AV I'm not sure what issue you are describing with the Series with Seasons and Episodes.  Have you clicked "Play all" to play all of the Episodes?  I have a test feed and it contains a Series with three Seasons with multiple Episodes.  It plays the next Episode after viewing the first one if I click 'Play all'.

**Reply by Play-AV** [reply]
What I was getting was initiating playback for a specific episode through the Episodes screen would often launch the 'first' episode of the first season as defined in the json for the series, regardless of which was selected for playback. I noticed the inconsistency since I had a series without a season (so just episodes within a series) and it worked as expected. 
I'll have to go back to my notes for specifics since it's been a month.

**Reply by rrirower** [reply]
@Play-AV Make sure you are using the latest version of the Wizard.  What you describe sounds like a bug that may have been previously resolved.

**Comment by ferdiworks** [comment]
@Play-AV You wrote: Out of curiosity, do you manually add in your deep linking for every item? No, you Deeplink by adding one item!

**Comment by Play-AV** [comment]
> @Play-AV You wrote: Out of curiosity, do you manually add in your deep linking for every item? No, you Deeplink by adding one item!

Okay great. It seems I was correct then. You do one per media type defined in your feed or just one overall?

**Reply by rrirower** [reply]
@Play-AV There's nothing to add to achieve deep-linking.  The code handles that request based on how the channel was started.

**Reply by ferdiworks** [reply]
@Play-AV On the Developer account side you have to go and deeplink one media item! The code as @rrirower stated There's nothing to add to achieve deep-linking. The code handles that request based on how the channel was started.

**Reply by adamesq** [reply]
Just confirming! I did as @ferdiworks suggested. I put in one item from each category (I use TV episodes and short vids). All is good.

**Reply by ferdiworks** [reply]
@adamesq You only needed to add one media item on the deep link in your developer account! That one item should be a video you plan on never deleting!

**Reply by adamesq** [reply]
Got it! Interesting. Thanx for the clarification!
